testworkflow.go

```diff
@@ -1049,8 +1053,22 @@ func parseResult(res testResult, localPath string) junit.Testsuite {
 		// Tests handled by a suite but not executed or skipped should be marked disabled
 		for _, test := range getTestsBySuiteName(res.testWorkflow.Name, localPath) {
 			hasResult := false
-			for _, tc := range ret.Testcases {
+			for idx, tc := range ret.Testcases {
```

Comment:
I like the flag, but there is a more direct way to skip these tests than modifying test results (this method would also create some inconsistencies in exit status from CIT and XML results content).

The test binary should be executed using the flag `-test.skip` and the value of `exclude_discrete_tests`. All of our current logic uses positive matching to select test cases so this is a good use of the negative match flag. It's a bit convoluted to pass the flag all the way there, but you can  add the `exclude_discrete_tests` string representation to the `TestWorkflow struct`, pass the value to the VM in `setInstanceTestMetadata` and then fetch it in `wrapper/main.go` and add it to `testArguments`.

Reply:
All done :saluting_face: